### PR TITLE
remove some redundant variable assignments

### PR DESCRIPTION
**Pull request by SimonCropp**
# Bug

Fixes: 

## Description

remove some redundant variable assignments

## PR Checklist

- [ ] Meaningful title, helpful description and a linked NuGet/Home issue
- [ ] Added tests
- [ ] Link to an issue or pull request to update docs if this PR changes settings, environment variables, new feature, etc.

### CODE REVIEW

**Review by Nigusu-Allehu — APPROVED** [review]
Thank you for your contributions! Out of curiosity and for learning purposes, did you identify these redundant assignments manually by reviewing all our code, or did you use a script to help with that? It seems like it would be a lot of effort to do manually, and I'm impressed either way!

**Comment by SimonCropp** [comment]
@Nigusu-Allehu i use a variety of techniques and tools. usually i just open random file and review them. often starting with the larger files. when i find a problem/improvement i work out some approach to finding any other instances of that problem/improvement. this approach can be varied: a test  that uses roslyn, a regex search, features provided by an ide, etc

in this case it was relatively easy given rider has a feature that will find  redundant variable assignments

File: src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectJsonProjectProvider.cs [hunk 1]
```diff
@@ -70,7 +70,7 @@ public NuGetProject TryCreateNuGetProject(
                 // Treat projects with project.json as build integrated projects
                 // Search for projectName.project.json first, then project.json
                 // If the name cannot be determined, search only for project.json
-                string projectJsonPath = null;
+                string projectJsonPath;
```
Comment (nkolev92) [inline src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Projects/ProjectJsonProjectProvider.cs:73]: I'd feel more strongly about accepting this PR if it was coupled with a rule change as well: https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0059.

**Review by nkolev92 — COMMENTED** [review]
This is one of those stylistic changes that while it's good, I think it shines best if there's a rule change.

**Pull request closed without merging**